benchmark/poseidon2/poseidon2_benchmark.cc

```diff
@@ -41,7 +41,6 @@ void Run(SimpleReporter& reporter, const Poseidon2Config& config, Fn horizen_fn,
     if (base::Contains(config.vendors(), Vendor::Plonky3())) {
       poseidon2_config = crypto::Poseidon2Config<Field>::CreateCustom(
           15, 7, 8, 13, math::GetPoseidon2BabyBearInternalShiftVector<15>());
-      CHECK_EQ(config.vendors().size(), static_cast<size_t>(1));
```

Comment:
Isn't it a right condition? The below constraint was if the field is `BabyBear`, the vendors should contain only `Plonky3`?

Reply:
Yes indeed! the allowed cases are [babybear: plonky3], [babybear: horizen], and [bn254: horizen + plonky3] (implemented in this order here); however, since the constraint below was implemented, this line 44 becomes redundant as you can see in line 40, line 44 simply ensures the case for [babybear: plonky3].